cornichon-core/src/test/scala/com/github/agourlay/cornichon/matchers/MatchersSpec.scala

```diff
@@ -68,5 +73,33 @@ class MatchersSpec extends WordSpec
         }
       }
     }
+
+    "any-date-time" must {
+      "run correctly all ISO-compliant values" in {
+        forAll(instantGen) { instant ⇒
+          println(instant)
+          anyDateTime.predicate(Json.fromString(DateTimeFormatter.ISO_INSTANT.format(instant))) should be(true)
+        }
+      }
+
+      "run correctly in parallel" in {
+        1.to(64)
+          .map(_ ⇒ reasonablyRandomInstantGen.sample)
+          .par
+          .foreach { instant ⇒
+            anyDateTime.predicate(Json.fromString(DateTimeFormatter.ISO_INSTANT.format(instant.get))) should be(true)
+          }
+      }
+    }
   }
 }
+
+object MatchersSpec {
+  val reasonablyRandomInstantGen: Gen[Instant] = for {
```

Comment:
why do we need two different `Gen[Instant]`?

Reply:
The pull request adresses two issues in fact:

1. matcher thread-safety
2. date formats allowed by the matcher

Perhaps it's a bit redundant, but having separate generators allows us to have separate test scenarios for both, highlighting the exact thing tested in each.

For the thread-safety scenario I used the "reasonable random" generator which produces dates somewhere between 1980s and 2020s. These yields timestamps in the regular yyyy-MM-dd* format that also passes with the previous SimpleDateFormat (format-wise).

For the ISO format scenario I used a generator that yields far more extreme dates, e.g. year 40'000. ISO dictates that the dates with years beyond 10'000 should be prepended with a `+`, which also causes the `SimpleDateFormat` to fail. See also https://tools.ietf.org/html/rfc2550 .